### DISCUSSION TITLE
Query Params available in SSG vs SSR

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#15466] [Help]
I am totally stuck on trying to understand a part of Next.js related to the params and router.query.

When I run the Post example from https://nextjs.org/docs/routing/dynamic-routes and print out the router.query object with `console.log(`query ${JSON.stringify(router.query)}`);`, I can see the route parameter and all query parameters in the browser console.

When I am working in a SSG with fallback=true, and I do the same thing, but print out the params object in getStaticProps with `console.log(`getStaticProps - Enter ${JSON.stringify(params)}`);`, during runtime when I hit a post that was generated during build time....all I see printed in the output is the url/route parameter...no query parameters are printed in the array.

Why the different behavior, what I am I missing?  How do you access query parameters from within getStaticProps?

**Comment by lucasbasquerotto** [comment]
When you use SSG the props are generated at build time instead of at request time, so `getStaticProps` wouldn't see the request as different from another with the same path, even if the query parameters are different, so it ignores the query parameters.

That said, it would be great if there was a way to define which query parameters it shouldn't ignore, so it would generate different pages if the specified query parameters are different (just like it already does with different slugs in the path).

Related: https://github.com/vercel/next.js/discussions/17616 and https://github.com/vercel/next.js/discussions/17529#discussioncomment-94228
